src/compiler/scala/tools/nsc/typechecker/Infer.scala

```diff
@@ -1146,11 +1132,23 @@ trait Infer extends Checkable {
     // (see #3692, where the type param T's bounds were set to > : T <: T, so that parts looped)
     // the changes are rolled back by restoreTypeBounds, but might be unintentionally observed in the mean time
     def instantiateTypeVar(tvar: TypeVar): Unit = {
-      val tparam                    = tvar.origin.typeSymbol
-      val TypeBounds(lo0, hi0)      = tparam.info.bounds
-      val tb @ TypeBounds(lo1, hi1) = instBounds(tvar)
-      val enclCase                  = context.enclosingCaseDef
-      def enclCase_s                = enclCase.toString.replaceAll("\\n", " ").take(60)
+      val tparam = tvar.origin.typeSymbol
+      val tparams = cloneSymbols(tvar.typeParams)
+      val targs = tparams.map(_.tpeHK)
+      val instType = if (!tvar.instValid) Nil else {
+        val inst = toOrigin(genPolyType(tparams, appliedType(tvar.inst, targs)))
```

Comment:
@adriaanm there was a gotcha here - we can't call `appliedType` if not `tvar.instValid` so I had to change the code a bit since you last reviewed.